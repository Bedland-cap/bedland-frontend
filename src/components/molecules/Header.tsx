import { HeaderBox, Logo } from "theme/Header.styled";
import { HeaderI } from "utils/interfaces/HeaderI";

export const Header = (props: HeaderI) => {
  const { variant } = props;
  return (
    <HeaderBox theme={{ variant }}>
      <Logo src={require("assets/Logo.png")} theme={{ variant }} />
    </HeaderBox>
  );
};
