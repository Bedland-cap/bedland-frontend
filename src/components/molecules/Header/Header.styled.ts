import styled from 'styled-components';

type TStyledHeader = {
  theme: { logged: boolean; color: string };
};

export const HeaderBox = styled.div<TStyledHeader>`
  align-items: center;
  background-color: ${({ theme }) => theme.color};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: row;
  gap: 0.625rem;
  height: 4.375rem;
  justify-content: ${({ theme }) =>
    theme.logged ? 'space-between' : 'center'};
  padding-right: 1.25rem;
  width: 100%;
`;

export const Logo = styled.img<TStyledHeader>`
  margin-left: ${({ theme }) => (theme.logged ? '2rem' : '0rem')};
`;
