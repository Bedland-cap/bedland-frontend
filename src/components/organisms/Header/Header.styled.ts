import styled from 'styled-components';
import { HEADER_HEIGHT } from 'utils/constans';
import { PaletteType } from 'theme/theme.types';

type HeaderProps = {
  isLoggedIn: boolean;
  palette: PaletteType;
};

export const HeaderBox = styled.div<HeaderProps>`
  align-items: center;
  background-color: ${({ palette }) => palette.light};
  display: flex;
  gap: 10px;
  height: ${HEADER_HEIGHT}px;
  justify-content: ${({ isLoggedIn }) => (!isLoggedIn ? 'center' : 'flex-start')};
  width: 100%;
`;
export default HeaderBox;
