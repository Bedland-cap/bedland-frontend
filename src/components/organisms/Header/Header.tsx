import LogoImg from 'assets/img/Logo.png';
import { useContext } from 'react';
import { useAppSelector } from 'store/hooks';
import { selectUserIsLoggedIn } from 'store/reducers/user_slice';
import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';
import { ThemeContext } from 'theme/ThemeContext';

type HeaderProps = {
  isLoggedIn: boolean;
  palette: PaletteType;
};

const HeaderBox = styled.div<HeaderProps>`
  align-items: center;
  background-color: ${({ palette }) => palette.light};
  display: sticky;
  gap: 10px;
  height: 90px;
  justify-content: ${({ isLoggedIn }) => (isLoggedIn ? 'center' : 'flex-start')};
  width: 100%;
  & > img {
    margin: ${({ isLoggedIn }) => (isLoggedIn === false ? '2rem' : '1rem')};
  }
`;

const Header = () => {
  const isLoggedIn: boolean = useAppSelector(selectUserIsLoggedIn);
  const { palette } = useContext(ThemeContext);
  return (
    <nav>
      <HeaderBox isLoggedIn={isLoggedIn} palette={palette}>
        <img src={LogoImg} alt='Logo Bedland' />
      </HeaderBox>
    </nav>
  );
};

export default Header;
