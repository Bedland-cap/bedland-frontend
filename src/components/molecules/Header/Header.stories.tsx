import type { Meta, StoryObj } from '@storybook/react';
import ThemeProvider from 'theme/ThemeContext';
import Header from './Header';

const meta = {
  title: 'molecules/Header',
  component: Header,
  tags: ['autodocs'],
  decorators: [(Story) => <ThemeProvider>{Story()}</ThemeProvider>],
} satisfies Meta<typeof Header>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ResidentLoggedOut: Story = {
  args: {
    variant: 'resident',
    logged: false,
  },
};

export const ResidentLoggedIn: Story = {
  args: {
    variant: 'resident',
    logged: true,
  },
};
export const ManagerLoggedOut: Story = {
  args: {
    variant: 'manager',
    logged: false,
  },
};

export const ManagerLoggedIn: Story = {
  args: {
    variant: 'manager',
    logged: true,
  },
};
