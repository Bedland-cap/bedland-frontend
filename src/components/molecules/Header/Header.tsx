import { HeaderBox, Logo } from "components/molecules/Header/Header.styled";
import { HeaderI } from "utils/interfaces/HeaderI";

export const Header = (props: HeaderI) => {
  const { variant } = props;
  return (
    <nav>
      <HeaderBox theme={{ variant }}>
        <Logo src={require("assets/Logo.png")} theme={{ variant }} />
      </HeaderBox>
    </nav>
  );
};
