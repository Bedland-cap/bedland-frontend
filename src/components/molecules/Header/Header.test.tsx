import renderer from 'react-test-renderer';
import ThemeProvider from 'theme/ThemeContext';
import Header from './Header';

describe('Header', () => {
  it('renders correctly if logged is false', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <Header variant='resident' logged={false} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly if logged is true', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <Header variant='resident' logged />
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
