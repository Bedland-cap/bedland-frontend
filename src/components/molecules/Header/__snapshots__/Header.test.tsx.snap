// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly if logged is false 1`] = `
<div
  className="sc-gswNZR iAYawr"
>
  <img
    className="sc-dkrFOg VIrMd"
  />
</div>
`;

exports[`Header renders correctly if logged is true 1`] = `
<div
  className="sc-gswNZR dCqFrD"
>
  <img
    className="sc-dkrFOg VIQMX"
  />
  <button
    className="sc-bcXHqe iJLHBU"
    type="button"
  >
    dropdown here
  </button>
</div>
`;
